### ISSUE TITLE
Variable interdependecy casues terraform plan to fail in 4.1.2 

### ISSUE THREAD

**Issue by dipeti** [terraform-aws-modules/terraform-aws-rds#386]
4.1.2 introduced a situation whereby running terraform plan is not possible because the local variable ` create_random_password`
```
create_random_password = var.create_db_instance && var.create_random_password && var.replicate_source_db == null
```
is evaluated based on the dynamic `var.replicate_source_db` whose value cannot be determined until after terraform apply. 

`replicate_source_db` is usually referenced by another db instance that is typically created in the same terraform job, example:  
`replicate_source_db = module.rds.db_instance_id`

Terraform cannot tell whether `module.rds.db_instance_id` will be null or not thus it fails to render a plan.

**Comment by robins** [comment]
**Note**: Getting a similar error on v4.1.2. Importantly if I rollback to v4.1.1 it works just fine.

Getting a similar error for `replica-postgres`:

#### Repro steps
```
terraform init
terraform plan
```

#### Error
```
Error: Invalid count argument
│
│   on ..\..\main.tf line 14, in resource "random_password" "master_password":
│   14:   count = local.create_random_password ? 1 : 0
│
│ The "count" value depends on resource attributes that cannot be determined until apply, so Terraform cannot predict how many instances will be   
│ created. To work around this, use the -target argument to first apply only the resources that the count depends on.
```

**Comment by antonbabenko** [comment]
This issue has been resolved in version 4.1.3 :tada:
